kubelet/datadog_checks/kubelet/kubelet.py

```diff
@@ -58,6 +68,45 @@
 log = logging.getLogger('collector')
 
 
+class ExpiredPodFilter(object):
+    """
+    Allows to filter old pods out of the podlist by providing a decoding hook
+    """
+    def __init__(self, cutoff_date):
+        self.expired_count = 0
+        self.cutoff_date = cutoff_date
+
+    def json_hook(self, pod):
+        # Not a pod (hook is called for all objects)
+        if 'metadata' not in pod or 'status' not in pod:
```

Comment:
why not using `kind` value to continue only with pod?

Reply:
Great idea, but kind is not a metadata field for each pod (there's only one toplevel `kind:PodList`) :/